virt/virt.py

```diff
@@ -77,6 +77,37 @@ def toDict(self):
         return d
 
 
+class Hypervisor(object):
+    """
+    A model for information about a hypervisor
+    """
+    def __init__(self, hypervisorId,  guestIds=[], name=None):
+        """
+        Create a new Hypervisor that will be sent to subscription manager
+
+        'hypervisorId': the unique identifier for this hypervisor
+
+        'guestIds': a list of Guests
+
+        'name': the hostname, if available
+        """
+        self.hypervisorId = hypervisorId
+        self.guestIds = guestIds
+        self.name = name
+
+    def toDict(self):
+        d = {
+            'hypervisorId': self.hypervisorId,
+            'guestIds': [g.toDict() for g in self.guestIds]
+        }
+        if self.name is not None:
+            d['name'] = self.name
+        return d
+
```

Comment:
We are also looking to allow fact usage [sockets and what not] for the hosts. Not sure how/when they are to be collected on this end, but we should build the means to send them.